File: Android/iOrder/app/src/main/java/hr/fer/oobl/iorder/iorder/injection/ComponentFactory.java
package hr.fer.oobl.iorder.iorder.injection;

import hr.fer.oobl.iorder.iorder.base.IOrderApplication;
import hr.fer.oobl.iorder.iorder.injection.activity.ActivityComponent;
import hr.fer.oobl.iorder.iorder.injection.activity.DaggerActivity;
import hr.fer.oobl.iorder.iorder.injection.application.ApplicationComponent;
import hr.fer.oobl.iorder.iorder.injection.fragment.DaggerFragment;
import hr.fer.oobl.iorder.iorder.injection.fragment.FragmentComponent;

public final class ComponentFactory {

    private ComponentFactory() {
    }

    public static ApplicationComponent createApplicationComponent(final IOrderApplication iOrderApplication) {
        return ApplicationComponent.Initializer.init(iOrderApplication);
    }

    public static ActivityComponent createActivityComponent(final DaggerActivity daggerActivity, final ApplicationComponent applicationComponent) {
        return ActivityComponent.Initializer.init(daggerActivity, applicationComponent);
    }

    public static FragmentComponent createFragmentComponent(final DaggerFragment daggerFragment, final ActivityComponent activityComponent) {
        return FragmentComponent.Initializer.init(daggerFragment, activityComponent);
    }
}

File: Android/iOrder/app/src/main/java/hr/fer/oobl/iorder/iorder/injection/ComponentFactory.kt
package hr.fer.oobl.iorder.iorder.injection


import hr.fer.oobl.iorder.iorder.application.IOrderApplication
import hr.fer.oobl.iorder.iorder.injection.activity.ActivityComponent
import hr.fer.oobl.iorder.iorder.injection.activity.DaggerActivity
import hr.fer.oobl.iorder.iorder.injection.application.ApplicationComponent
import hr.fer.oobl.iorder.iorder.injection.fragment.DaggerFragment
import hr.fer.oobl.iorder.iorder.injection.fragment.FragmentComponent

object ComponentFactory {

    fun createApplicationComponent(iOrderApplication: IOrderApplication): ApplicationComponent {
        return ApplicationComponent.Initializer.init(iOrderApplication)
    }

    fun createActivityComponent(daggerActivity: DaggerActivity, applicationComponent: ApplicationComponent): ActivityComponent {
        return ActivityComponent.Initializer.init(daggerActivity, applicationComponent)
    }

    fun createFragmentComponent(daggerFragment: DaggerFragment, activityComponent: ActivityComponent): FragmentComponent {
        return FragmentComponent.Initializer.init(daggerFragment, activityComponent)
    }
}
